digest.go

```diff
@@ -69,24 +69,18 @@ func (c digestCache) Swap(i, j int) {
 	c[i], c[j] = c[j], c[i]
 }
 
-// Purge removes count oldest entries from DigestAuth.clients
-func (da *DigestAuth) Purge(count int) {
+// Purge removes all clients from DigestAuth.clients, as long as those clients
+// were last seen more than DigestAuth.ClientCacheTTL amount of time ago.
+func (da *DigestAuth) Purge() {
 	da.mutex.Lock()
 	defer da.mutex.Unlock()
-	entries := make([]digestCacheEntry, 0, len(da.clients))
-	for nonce, client := range da.clients {
-		entries = append(entries, digestCacheEntry{nonce, client.lastSeen})
-	}
-	cache := digestCache(entries)
-	sort.Sort(cache)
-
-	limit := count
-	if len(cache) < count {
-		limit = len(cache) - 1
-	}
 
-	for _, client := range cache[:limit] {
-		delete(da.clients, client.nonce)
+	for nonce, client := range da.clients {
```

Comment:
One more thought — if `ClientCacheTTL` is zero (unset) then can we default to exact same old behavior? So, removing `2 * ClientCacheTolerance` entries each time (rather than all entries)?

This way, anyone using this package will find that `Purge` continues to work exactly as they would expect. Only if someone explicitly configures `ClientCacheTTL` will we purge entries according to this new TTL scheme.

We'd need to update comments in this file to explain this as well.

Reply:
Updated, although just for clarity we may purge less than `2 * ClientCacheTolerance` if there are not enough entries.

Reply:
Yes, thanks for the clarification.